### DISCUSSION TITLE
Epson M2170 PPD driver

### DISCUSSION THREAD

**Discussion by Armandt88** [OpenPrinting/cups#289] [Ideas] [answered]
Good day,

We are in the process of deploying Epson M2170 printers but the issue is we are running Odoo on Ubuntu and therefor need a CUPS PPD driver for the printers. Is this something that you will be willing to assist us with or point us towards a PPD that will do the job?

Thank you in advance.

**Comment by zdohnal** [comment, marked as answer]
Hi,

the classic PPD drivers are deprecated and their support in CUPS will go away with CUPS 3.0, but there are newer ways for printer support on Linux:

- If your printer supports AirPrint/IPP Everywhere (check your device user manual/specs - for network connection you need to open IPP port. For USB you need to install additional package - ipp-usb), use IPP Everywhere driver ('IPP Everywhere' driver in CUPS web ui, or '-m everywhere' via lpadmin). 
- If the device doesn't support the mentioned standards, give our printer applications a try (projects which ends with -printer-app [here](https://github.com/OpenPrinting/)) - they are available as snaps, some distros will package/packages them as well.

Further details heavily depends on deployment... if you deploy a server, you need to install the queue permanently, if only clients and the printer or sharing server is in the same subnet, then you don't need to install a queue at all - just install and start avahi - this will show printers in any print dialog which works with new (approx. 5 years old) CUPS API - otherwise you can start cups-browsed to automatically install the queues or install them manually...

**Reply by Armandt88** [reply]
Hi @zdohnal 

Thank you so much, it seems everything picked up automatically on Connection: | ipps://xxxxxx.local:631/ipp/print . All that was needed was a reboot. Thank you for the amazing product and hard work.
